src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -505,7 +505,20 @@ class Address {
   }
 
   bool uses(Register reg) const {
-    return base() == reg || index() == reg;
+    switch (_mode) {
+    case literal:
```

Comment:
Nit: You might want to leave a two-space indentation for each case. I think that's the code style :-)

Reply:
Yeah, indeed. I even noticed this inconsistency of coding style in hotspot shared code. I guess it's OK to align with the surrounding code. Looks good to me. PS: I thik I will prepare a similar enhancement for RISC-V on JDK master.

Reply:
> ```
> bool uses(Register reg) const {
>   return _mode != literal && (base() == reg || index == reg);
> }
> ```

Mmm, but that's fragile under change too, e.g. if we get another literal type. I guess you can argue that's never likely to happen, but I don't think it's appropriate to choose a less robust solution now.

Reply:
If a new literal type were to show up, there are many places that would need to be examined and probably
updated, including here, because of the explicit dispatch on all modes.  So I don't think what I'm suggesting
is any less robust than the current dispatch.  (Note that any change here should be for 21 - my suggested
change doesn't make the cut for 20 at this point.)